timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py

```diff
@@ -64,13 +80,17 @@ def __iter__(self) -> Iterator[Dict[str, Any]]:
             df = self.time_series_df.loc[item_id]
             time_series = {
                 FieldName.ITEM_ID: item_id,
-                FieldName.TARGET: df[self.target_field_name].to_numpy(),
+                FieldName.TARGET: df[self.target_field_name].to_numpy(dtype=self.float_dtype),
                 FieldName.START: pd.Period(df.index[0], freq=self.freq),
             }
             if self.feat_static_cat is not None:
-                time_series[FieldName.FEAT_STATIC_CAT] = self.feat_static_cat.loc[item_id].to_numpy()
+                time_series[FieldName.FEAT_STATIC_CAT] = self.feat_static_cat.loc[item_id].to_numpy(
+                    dtype=self.int_dtype
```

Comment:
We already did conversion to the correct float/int dtype in `__init__`, this seems redundant?